### DISCUSSION TITLE
Weekly Status: Week 29 2021

### DISCUSSION THREAD

**Discussion by cwayne18** [k3s-io/k3s#3708] [Weekly Status]
# Weekly Report
Weekly status report for k3s-io/k3s Week #29 2021

## Here's what the team has focused on this week:
* Bumped containerd to fix embargoed CVE 
* Fixed CVE-2021-32001 ([details here](https://github.com/k3s-io/k3s/security/advisories/GHSA-cxm9-4m6p-24mc))
* Released k3s 
  - [v1.21.3+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.21.3%2Bk3s1)
  - [v1.20.9+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.20.9%2Bk3s1)
  - [v1.19.13+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.19.13%2Bk3s1)
* PR opened for new integration test framework
* Ongoing process improvement retrospective
* Updated CI to not run when non-code changes are made [3676](https://github.com/k3s-io/k3s/pull/3676)
* Updated Documentation
  - [k3s network policy for traefik ingress](https://github.com/rancher/docs/pull/3409)

## Weekly Stats
| | Opened this week| Closed this week|
|--|---|-----|
|Issues|  53 |  51|
|PR's|  52 |  61|

|  |  |
|--|--|
| New stars | 97|
| New forks | 2|

## PR's Closed
* [#3701](https://github.com/k3s-io/k3s/pull/3701):  Update K3S 1.21 stable version
* [#3697](https://github.com/k3s-io/k3s/pull/3697):  [hotfix-1.19.12] Bump containerd to v1.4.8-k3s1
* [#3696](https://github.com/k3s-io/k3s/pull/3696):  TEST Update Readme DO NOT MERGE
* [#3694](https://github.com/k3s-io/k3s/pull/3694):  [hotfix-1.20.8] Bump containerd to v1.4.8-k3s1
* [#3693](https://github.com/k3s-io/k3s/pull/3693):  [hotfix-1.21.2] Bump containerd to v1.4.8-k3s1
* [#3692](https://github.com/k3s-io/k3s/pull/3692):  [Release-1.19] Fix multiple bootstrap keys found
* [#3691](https://github.com/k3s-io/k3s/pull/3691):  [Release-1.20] Fix multiple bootstrap keys found
* [#3690](https://github.com/k3s-io/k3s/pull/3690):  [Release-1.21] Fix multiple bootstrap keys found
* [#3688](https://github.com/k3s-io/k3s/pull/3688):  Fix multiple bootstrap keys found
* [#3687](https://github.com/k3s-io/k3s/pull/3687):  move go routines for api server ready beneath wait group
* [#3686](https://github.com/k3s-io/k3s/pull/3686):  Bump Kubernetes to v1.21.3
* [#3685](https://github.com/k3s-io/k3s/pull/3685):  [release-1.19] Bump containerd to v1.4.8-k3s1
* [#3684](https://github.com/k3s-io/k3s/pull/3684):  [release-1.20] Bump containerd to v1.4.8-k3s1
* [#3683](https://github.com/k3s-io/k3s/pull/3683):  [release-1.21] Bump containerd to v1.4.8-k3s1
* [#3682](https://github.com/k3s-io/k3s/pull/3682):  Bump containerd to v1.4.8-k3s1
* [#3676](https://github.com/k3s-io/k3s/pull/3676):  Ignore markdown files for github actions
* [#3674](https://github.com/k3s-io/k3s/pull/3674):  adding startup hooks args to access to Disables and Skips
* [#3673](https://github.com/k3s-io/k3s/pull/3673):  [Backport 1.20] Fix to allow prune to correctly cleanup custom named snapshots (#3649)
* [#3672](https://github.com/k3s-io/k3s/pull/3672):  [Backport 1.21] Fix to allow prune to correctly cleanup custom named snapshots (#3649)
* [#3669](https://github.com/k3s-io/k3s/pull/3669):  Add checkbox to denote backporting required on issue templates
* [#3668](https://github.com/k3s-io/k3s/pull/3668):  [release-1.19] Backport empty string fix
* [#3667](https://github.com/k3s-io/k3s/pull/3667):  [release-1.19] Upgrade k3s-root version
* [#3666](https://github.com/k3s-io/k3s/pull/3666):  [release-1.20] Upgrade k3s-root version
* [#3665](https://github.com/k3s-io/k3s/pull/3665):  [release-1.21] Upgrade k3s-root version
* [#3664](https://github.com/k3s-io/k3s/pull/3664):  Revert "Adding support for waitgroup to the Startuphooks (#3654)
* [#3663](https://github.com/k3s-io/k3s/pull/3663):  Add a checkbox notating required backports in PULL_REQUEST_TEMPLATE.md
* [#3659](https://github.com/k3s-io/k3s/pull/3659):  Finding a best fit acronym explanation
* [#3658](https://github.com/k3s-io/k3s/pull/3658):  Adding support for waitgroup to the Startuphooks (#3654)
* [#3656](https://github.com/k3s-io/k3s/pull/3656):  [release-1.19] Bump Kubernetes to v1.19.13
* [#3655](https://github.com/k3s-io/k3s/pull/3655):  [release-1.20] Bump Kubernetes to v1.20.9
* [#3654](https://github.com/k3s-io/k3s/pull/3654):  Adding support for waitgroup to the Startuphooks
* [#3652](https://github.com/k3s-io/k3s/pull/3652):  [release-1.21] Bump Kubernetes to v1.21.3
* [#3650](https://github.com/k3s-io/k3s/pull/3650):  [Release-1.20] Fix cluster reset call
* [#3649](https://github.com/k3s-io/k3s/pull/3649):  Fix to allow prune to correctly cleanup custom named snapshots
* [#3647](https://github.com/k3s-io/k3s/pull/3647):  [Release-1.20] Fix delete issue for etcd s3 snapshots
* [#3644](https://github.com/k3s-io/k3s/pull/3644):  Bump helm-controller to v0.10.1
* [#3627](https://github.com/k3s-io/k3s/pull/3627):  fix a runtime core panic
* [#3626](https://github.com/k3s-io/k3s/pull/3626):  Upgrade k3s-root version
* [#3623](https://github.com/k3s-io/k3s/pull/3623):  Backport issue 3414 release 1.20 2
* [#3621](https://github.com/k3s-io/k3s/pull/3621):  Convert existing unit tests to standard layout
* [#3615](https://github.com/k3s-io/k3s/pull/3615):  [release-1.21] Add events to deploy controller
* [#3614](https://github.com/k3s-io/k3s/pull/3614):  [release-1.21] Add nodename to UA string for deploy controller
* [#3611](https://github.com/k3s-io/k3s/pull/3611):  [Backport 1.20] - prevent snapshot save when snapshots are disabled (#3475)
* [#3610](https://github.com/k3s-io/k3s/pull/3610):  [Backport 1.21] - prevent snapshot save when snapshots are disabled (#3475)
* [#3604](https://github.com/k3s-io/k3s/pull/3604):  Add issue template for creating release checklist issues
* [#3603](https://github.com/k3s-io/k3s/pull/3603):  [release-1.21] Bump the packaged runc binary version
* [#3602](https://github.com/k3s-io/k3s/pull/3602):  Bump the packaged runc binary version
* [#3599](https://github.com/k3s-io/k3s/pull/3599):  [release-1.20] Sign CSRs for kubelet-serving with the server CA
* [#3592](https://github.com/k3s-io/k3s/pull/3592):  [release-1.21] Update etcd snapshot error message to be more informative
* [#3541](https://github.com/k3s-io/k3s/pull/3541):  [release-1.20] Update coredns
* [#3539](https://github.com/k3s-io/k3s/pull/3539):  [release-1.19] Update coredns
* [#3516](https://github.com/k3s-io/k3s/pull/3516):  [release-1.20] - Backport K3s/RKE2 bootstrap data is encrypted with empty string if user does not supply a token
* [#3514](https://github.com/k3s-io/k3s/pull/3514):  [release-1.21] - Backport Fix storing bootstrap data with empty token string
* [#3483](https://github.com/k3s-io/k3s/pull/3483):  Update tcpproxy upstream from github.com/google/tcpproxy to inet.af/tcpproxy
* [#3475](https://github.com/k3s-io/k3s/pull/3475):  prevent snapshot save when snapshots are disabled
* [#3140](https://github.com/k3s-io/k3s/pull/3140):  Add restoration from rke1 snapshot
* [#2897](https://github.com/k3s-io/k3s/pull/2897):  Fixed a typo in the README.md
* [#2803](https://github.com/k3s-io/k3s/pull/2803):  Merge kubeconfigs to ~/.kube/config
* [#2761](https://github.com/k3s-io/k3s/pull/2761):  Do not set KUBECONFIG env in kubectl CLI
* [#2141](https://github.com/k3s-io/k3s/pull/2141):  Updated Readme
* [#1935](https://github.com/k3s-io/k3s/pull/1935):  Add ability to disable kubeproxy

## PR's Opened
* [#3709](https://github.com/k3s-io/k3s/pull/3709):  add gotests templates
* [#3701](https://github.com/k3s-io/k3s/pull/3701):  Update K3S 1.21 stable version
* [#3697](https://github.com/k3s-io/k3s/pull/3697):  [hotfix-1.19.12] Bump containerd to v1.4.8-k3s1
* [#3696](https://github.com/k3s-io/k3s/pull/3696):  TEST Update Readme DO NOT MERGE
* [#3695](https://github.com/k3s-io/k3s/pull/3695):  Introduction of Integration Tests
* [#3694](https://github.com/k3s-io/k3s/pull/3694):  [hotfix-1.20.8] Bump containerd to v1.4.8-k3s1
* [#3693](https://github.com/k3s-io/k3s/pull/3693):  [hotfix-1.21.2] Bump containerd to v1.4.8-k3s1
* [#3692](https://github.com/k3s-io/k3s/pull/3692):  [Release-1.19] Fix multiple bootstrap keys found
* [#3691](https://github.com/k3s-io/k3s/pull/3691):  [Release-1.20] Fix multiple bootstrap keys found
* [#3690](https://github.com/k3s-io/k3s/pull/3690):  [Release-1.21] Fix multiple bootstrap keys found
* [#3688](https://github.com/k3s-io/k3s/pull/3688):  Fix multiple bootstrap keys found
* [#3687](https://github.com/k3s-io/k3s/pull/3687):  move go routines for api server ready beneath wait group
* [#3686](https://github.com/k3s-io/k3s/pull/3686):  Bump Kubernetes to v1.21.3
* [#3685](https://github.com/k3s-io/k3s/pull/3685):  [release-1.19] Bump containerd to v1.4.8-k3s1
* [#3684](https://github.com/k3s-io/k3s/pull/3684):  [release-1.20] Bump containerd to v1.4.8-k3s1
* [#3683](https://github.com/k3s-io/k3s/pull/3683):  [release-1.21] Bump containerd to v1.4.8-k3s1
* [#3682](https://github.com/k3s-io/k3s/pull/3682):  Bump containerd to v1.4.8-k3s1
* [#3676](https://github.com/k3s-io/k3s/pull/3676):  Ignore markdown files for github actions
* [#3674](https://github.com/k3s-io/k3s/pull/3674):  adding startup hooks args to access to Disables and Skips
* [#3673](https://github.com/k3s-io/k3s/pull/3673):  [Backport 1.20] Fix to allow prune to correctly cleanup custom named snapshots (#3649)
* [#3672](https://github.com/k3s-io/k3s/pull/3672):  [Backport 1.21] Fix to allow prune to correctly cleanup custom named snapshots (#3649)
* [#3669](https://github.com/k3s-io/k3s/pull/3669):  Add checkbox to denote backporting required on issue templates
* [#3668](https://github.com/k3s-io/k3s/pull/3668):  [release-1.19] Backport empty string fix
* [#3667](https://github.com/k3s-io/k3s/pull/3667):  [release-1.19] Upgrade k3s-root version
* [#3666](https://github.com/k3s-io/k3s/pull/3666):  [release-1.20] Upgrade k3s-root version
* [#3665](https://github.com/k3s-io/k3s/pull/3665):  [release-1.21] Upgrade k3s-root version
* [#3664](https://github.com/k3s-io/k3s/pull/3664):  Revert "Adding support for waitgroup to the Startuphooks (#3654)
* [#3663](https://github.com/k3s-io/k3s/pull/3663):  Add a checkbox notating required backports in PULL_REQUEST_TEMPLATE.md
* [#3662](https://github.com/k3s-io/k3s/pull/3662):  Added logic to strip any existing hyphens before processing the args.
* [#3661](https://github.com/k3s-io/k3s/pull/3661):  Exporting the AddFeatureGate function and adding a unit test for it.
* [#3659](https://github.com/k3s-io/k3s/pull/3659):  Finding a best fit acronym explanation
* [#3658](https://github.com/k3s-io/k3s/pull/3658):  Adding support for waitgroup to the Startuphooks (#3654)
* [#3656](https://github.com/k3s-io/k3s/pull/3656):  [release-1.19] Bump Kubernetes to v1.19.13
* [#3655](https://github.com/k3s-io/k3s/pull/3655):  [release-1.20] Bump Kubernetes to v1.20.9
* [#3654](https://github.com/k3s-io/k3s/pull/3654):  Adding support for waitgroup to the Startuphooks
* [#3652](https://github.com/k3s-io/k3s/pull/3652):  [release-1.21] Bump Kubernetes to v1.21.3
* [#3650](https://github.com/k3s-io/k3s/pull/3650):  [Release-1.20] Fix cluster reset call
* [#3649](https://github.com/k3s-io/k3s/pull/3649):  Fix to allow prune to correctly cleanup custom named snapshots
* [#3647](https://github.com/k3s-io/k3s/pull/3647):  [Release-1.20] Fix delete issue for etcd s3 snapshots
* [#3644](https://github.com/k3s-io/k3s/pull/3644):  Bump helm-controller to v0.10.1
* [#3627](https://github.com/k3s-io/k3s/pull/3627):  fix a runtime core panic
* [#3626](https://github.com/k3s-io/k3s/pull/3626):  Upgrade k3s-root version
* [#3623](https://github.com/k3s-io/k3s/pull/3623):  Backport issue 3414 release 1.20 2
* [#3621](https://github.com/k3s-io/k3s/pull/3621):  Convert existing unit tests to standard layout
* [#3615](https://github.com/k3s-io/k3s/pull/3615):  [release-1.21] Add events to deploy controller
* [#3614](https://github.com/k3s-io/k3s/pull/3614):  [release-1.21] Add nodename to UA string for deploy controller
* [#3611](https://github.com/k3s-io/k3s/pull/3611):  [Backport 1.20] - prevent snapshot save when snapshots are disabled (#3475)
* [#3610](https://github.com/k3s-io/k3s/pull/3610):  [Backport 1.21] - prevent snapshot save when snapshots are disabled (#3475)
* [#3604](https://github.com/k3s-io/k3s/pull/3604):  Add issue template for creating release checklist issues
* [#3603](https://github.com/k3s-io/k3s/pull/3603):  [release-1.21] Bump the packaged runc binary version
* [#3602](https://github.com/k3s-io/k3s/pull/3602):  Bump the packaged runc binary version
* [#3599](https://github.com/k3s-io/k3s/pull/3599):  [release-1.20] Sign CSRs for kubelet-serving with the server CA

## Issues Opened
* [#3710](https://github.com/k3s-io/k3s/issues/3710):  Update Documentation for Unit Tests 
* [#3706](https://github.com/k3s-io/k3s/issues/3706):  Base level configuration for gotests
* [#3705](https://github.com/k3s-io/k3s/issues/3705):  Can /etc/rancher/k3s/config.yaml be used to init k3s-agents?
* [#3704](https://github.com/k3s-io/k3s/issues/3704):  Cannot write data to local PVC
* [#3703](https://github.com/k3s-io/k3s/issues/3703):  k3s issue with helm3 and pvc pending
* [#3702](https://github.com/k3s-io/k3s/issues/3702):  Update SECURITY.md
* [#3700](https://github.com/k3s-io/k3s/issues/3700):  Fedora VM Worker Not Joining
* [#3699](https://github.com/k3s-io/k3s/issues/3699):  k3s service failed to restart after etcd error
* [#3698](https://github.com/k3s-io/k3s/issues/3698):  Promote v1.21.3+k3s1 to stable
* [#3689](https://github.com/k3s-io/k3s/issues/3689):  failed to get CA certs (with public IP)
* [#3681](https://github.com/k3s-io/k3s/issues/3681):  [release-1.19] Bump containerd to v1.4.8-k3s1
* [#3680](https://github.com/k3s-io/k3s/issues/3680):  [release-1.20] Bump containerd to v1.4.8-k3s1
* [#3679](https://github.com/k3s-io/k3s/issues/3679):  [release-1.21] Bump containerd to v1.4.8-k3s1
* [#3678](https://github.com/k3s-io/k3s/issues/3678):  Bump containerd to v1.4.8-k3s1
* [#3675](https://github.com/k3s-io/k3s/issues/3675):  K3s/RKE2 bootstrap data is encrypted with empty string if user does not supply a token
* [#3671](https://github.com/k3s-io/k3s/issues/3671):  [release 1.20] Etcd snapshot prune fails to delete non default named snapshots
* [#3670](https://github.com/k3s-io/k3s/issues/3670):  [release 1.21] Etcd snapshot prune fails to delete non default named snapshots
* [#3660](https://github.com/k3s-io/k3s/issues/3660):  Massive state.db-wal file
* [#3653](https://github.com/k3s-io/k3s/issues/3653):  Adding support for waitgroup to the Startuphooks
* [#3651](https://github.com/k3s-io/k3s/issues/3651):  Passing cluster-reset fails to trigger reset due to force-new-cluster value set to false
* [#3648](https://github.com/k3s-io/k3s/issues/3648):  Etcd snapshot prune fails to delete non default named snapshots.
* [#3646](https://github.com/k3s-io/k3s/issues/3646):  Unable to delete snapshot from s3 bucket using etcd snapshot delete subcommand
* [#3643](https://github.com/k3s-io/k3s/issues/3643):  Make the addFeatureGate function an exported function.
* [#3642](https://github.com/k3s-io/k3s/issues/3642):  GetArgsList function appends additional hyphens to args if it already has hyphens.
* [#3639](https://github.com/k3s-io/k3s/issues/3639):  Docker & registries.yaml does not work for GCR
* [#3638](https://github.com/k3s-io/k3s/issues/3638):  Does K3S support the production environment of traditional Internet companies (not IoT or edge computing)?
* [#3637](https://github.com/k3s-io/k3s/issues/3637):  K3S - Service can not start
* [#3636](https://github.com/k3s-io/k3s/issues/3636):  While running rootless node password file path is incorrect
* [#3635](https://github.com/k3s-io/k3s/issues/3635):  [release-1.19] Upgrade k3s-root to 0.9.1 version
* [#3634](https://github.com/k3s-io/k3s/issues/3634):  [release-1.20] Upgrade k3s-root to 0.9.1 version
* [#3633](https://github.com/k3s-io/k3s/issues/3633):  [release-1.21] Upgrade k3s-root to 0.9.1 version
* [#3632](https://github.com/k3s-io/k3s/issues/3632):  [Release-1.21] Restoration on a fresh node fails to verify hash for node using the node password file
* [#3631](https://github.com/k3s-io/k3s/issues/3631):  [Release-1.20] Restoration on a fresh node fails to verify hash for node using the node password file
* [#3630](https://github.com/k3s-io/k3s/issues/3630):  [release-1.20] Bump kine to resolve race condition and unrevisioned delete
* [#3629](https://github.com/k3s-io/k3s/issues/3629):  [release-1.21] Bump kine to resolve race condition and unrevisioned delete
* [#3628](https://github.com/k3s-io/k3s/issues/3628):  K3S unauthorized errors - failing to start
* [#3625](https://github.com/k3s-io/k3s/issues/3625):  Upgrade k3s-root to 0.9.1 version
* [#3624](https://github.com/k3s-io/k3s/issues/3624):  DNS resolution stop working after k3s installation
* [#3622](https://github.com/k3s-io/k3s/issues/3622):  Allow executor to pass additional parameters for customization
* [#3620](https://github.com/k3s-io/k3s/issues/3620):  Failed to prepare subPath for volumeMount of the container --rootless
* [#3619](https://github.com/k3s-io/k3s/issues/3619):  Expose kube-scheduler, kube-proxy and kube-controller metrics endpoints
* [#3618](https://github.com/k3s-io/k3s/issues/3618):  k3s kube-proxy
* [#3617](https://github.com/k3s-io/k3s/issues/3617):  K3S secret-encryption feature enhancement to support  - decrypt all secrets, Rotating a decryption key
* [#3613](https://github.com/k3s-io/k3s/issues/3613):  [release-1.21] Use managedFields to track which node's deploy controller modified a resource
* [#3612](https://github.com/k3s-io/k3s/issues/3612):  [release-1.21] Improve feedback of deploy controller operations and failures
* [#3609](https://github.com/k3s-io/k3s/issues/3609):  [Backport 1.19] - prevent snapshot save when snapshots are disabled
* [#3608](https://github.com/k3s-io/k3s/issues/3608):  [Backport 1.20] - prevent snapshot save when snapshots are disabled
* [#3607](https://github.com/k3s-io/k3s/issues/3607):  [Backport 1.21] - prevent snapshot save when snapshots are disabled
* [#3601](https://github.com/k3s-io/k3s/issues/3601):  [release-1.21] Bump runc to v1.0.0
* [#3600](https://github.com/k3s-io/k3s/issues/3600):  Bump runc to v1.0.0
* [#3598](https://github.com/k3s-io/k3s/issues/3598):  Traefik consume lots of RAM and CPU power, k3s hangs. Occured after setting the externalName
* [#3597](https://github.com/k3s-io/k3s/issues/3597):  Unable to add new master to cluster with embedded etcd
* [#3596](https://github.com/k3s-io/k3s/issues/3596):  rootless: remove cgroup v1 support ("none" cgroup driver)

## Issues Closed
* [#3698](https://github.com/k3s-io/k3s/issues/3698):  Promote v1.21.3+k3s1 to stable
* [#3681](https://github.com/k3s-io/k3s/issues/3681):  [release-1.19] Bump containerd to v1.4.8-k3s1
* [#3680](https://github.com/k3s-io/k3s/issues/3680):  [release-1.20] Bump containerd to v1.4.8-k3s1
* [#3679](https://github.com/k3s-io/k3s/issues/3679):  [release-1.21] Bump containerd to v1.4.8-k3s1
* [#3675](https://github.com/k3s-io/k3s/issues/3675):  K3s/RKE2 bootstrap data is encrypted with empty string if user does not supply a token
* [#3651](https://github.com/k3s-io/k3s/issues/3651):  Passing cluster-reset fails to trigger reset due to force-new-cluster value set to false
* [#3646](https://github.com/k3s-io/k3s/issues/3646):  Unable to delete snapshot from s3 bucket using etcd snapshot delete subcommand
* [#3639](https://github.com/k3s-io/k3s/issues/3639):  Docker & registries.yaml does not work for GCR
* [#3638](https://github.com/k3s-io/k3s/issues/3638):  Does K3S support the production environment of traditional Internet companies (not IoT or edge computing)?
* [#3637](https://github.com/k3s-io/k3s/issues/3637):  K3S - Service can not start
* [#3632](https://github.com/k3s-io/k3s/issues/3632):  [Release-1.21] Restoration on a fresh node fails to verify hash for node using the node password file
* [#3631](https://github.com/k3s-io/k3s/issues/3631):  [Release-1.20] Restoration on a fresh node fails to verify hash for node using the node password file
* [#3630](https://github.com/k3s-io/k3s/issues/3630):  [release-1.20] Bump kine to resolve race condition and unrevisioned delete
* [#3629](https://github.com/k3s-io/k3s/issues/3629):  [release-1.21] Bump kine to resolve race condition and unrevisioned delete
* [#3619](https://github.com/k3s-io/k3s/issues/3619):  Expose kube-scheduler, kube-proxy and kube-controller metrics endpoints
* [#3613](https://github.com/k3s-io/k3s/issues/3613):  [release-1.21] Use managedFields to track which node's deploy controller modified a resource
* [#3612](https://github.com/k3s-io/k3s/issues/3612):  [release-1.21] Improve feedback of deploy controller operations and failures
* [#3609](https://github.com/k3s-io/k3s/issues/3609):  [Backport 1.19] - prevent snapshot save when snapshots are disabled
* [#3608](https://github.com/k3s-io/k3s/issues/3608):  [Backport 1.20] - prevent snapshot save when snapshots are disabled
* [#3607](https://github.com/k3s-io/k3s/issues/3607):  [Backport 1.21] - prevent snapshot save when snapshots are disabled
* [#3601](https://github.com/k3s-io/k3s/issues/3601):  [release-1.21] Bump runc to v1.0.0
* [#3598](https://github.com/k3s-io/k3s/issues/3598):  Traefik consume lots of RAM and CPU power, k3s hangs. Occured after setting the externalName
* [#3593](https://github.com/k3s-io/k3s/issues/3593):  Traefik not reloading after a configuration update
* [#3591](https://github.com/k3s-io/k3s/issues/3591):  [Backport 1.21] etcd-snapshot needs to be run as root
* [#3583](https://github.com/k3s-io/k3s/issues/3583):  Rancher UI inaccessible with default hardened k3s settings
* [#3574](https://github.com/k3s-io/k3s/issues/3574):  Containerd mount service unit entered the 'dead' state on raspberry pi 4 (64-bit)
* [#3556](https://github.com/k3s-io/k3s/issues/3556):  Cut 1.21.3
* [#3555](https://github.com/k3s-io/k3s/issues/3555):  Cut 1.20.9
* [#3554](https://github.com/k3s-io/k3s/issues/3554):  Cut 1.19.13
* [#3547](https://github.com/k3s-io/k3s/issues/3547):  [Release 1.21] /var/lib/rancher/k3s/storage should not be world-readable
* [#3546](https://github.com/k3s-io/k3s/issues/3546):  [Backport 1.20] - Etcd Commands 
* [#3543](https://github.com/k3s-io/k3s/issues/3543):  [release-1.19] Update coredns to 1.8.3
* [#3542](https://github.com/k3s-io/k3s/issues/3542):  [release-1.20] Update coredns to 1.8.3
* [#3529](https://github.com/k3s-io/k3s/issues/3529):  [Backport 1.21] Move cloud-controller-manager functionality into dedicated controller
* [#3528](https://github.com/k3s-io/k3s/issues/3528):  [Backport 1.20] Move cloud-controller-manager functionality into dedicated controller
* [#3508](https://github.com/k3s-io/k3s/issues/3508):  Consoliated unit test asserts
* [#3503](https://github.com/k3s-io/k3s/issues/3503):  Test and Certify Rocky Linux 8.3 and 8.4 [k3s]
* [#3501](https://github.com/k3s-io/k3s/issues/3501):  Test and Certify OpenSUSE (Leap) 15 SP3 [k3s]
* [#3496](https://github.com/k3s-io/k3s/issues/3496):  Improve unit test coverage in pkg/etcd
* [#3485](https://github.com/k3s-io/k3s/issues/3485):  [Backport 1.21] - Possible race where bootstrap data might not save
* [#3484](https://github.com/k3s-io/k3s/issues/3484):  [Backport 1.20] - Possible race where bootstrap data might not save
* [#3446](https://github.com/k3s-io/k3s/issues/3446):  [release-1.20] CSRs for kubernetes.io/kubelet-serving are signed with the client CA instead of the server CA
* [#3444](https://github.com/k3s-io/k3s/issues/3444):  Added code coverage info to CI/Drone
* [#3427](https://github.com/k3s-io/k3s/issues/3427):  [release-1.21] Update helm-controller to v0.10.0 or newer
* [#3414](https://github.com/k3s-io/k3s/issues/3414):  Possible race where bootstrap data might not save
* [#3374](https://github.com/k3s-io/k3s/issues/3374):  [release-1.21] Update embedded kube-router network policy controller
* [#3228](https://github.com/k3s-io/k3s/issues/3228):  cgroups v2 validation
* [#2697](https://github.com/k3s-io/k3s/issues/2697):  Move cloud-controller-manager functionality into dedicated controller
* [#2539](https://github.com/k3s-io/k3s/issues/2539):  Supported OCI complaint container runtimes for K3s
* [#2348](https://github.com/k3s-io/k3s/issues/2348):  /var/lib/rancher/k3s/storage should not be world-readable.
* [#882](https://github.com/k3s-io/k3s/issues/882):  Remotedialer proxy error when using Google Cloud VM as k3s server

## Community PRs Closed
* [#3659](https://github.com/k3s-io/k3s/pull/3659):  Finding a best fit acronym explanation
* [#3483](https://github.com/k3s-io/k3s/pull/3483):  Update tcpproxy upstream from github.com/google/tcpproxy to inet.af/tcpproxy
* [#2897](https://github.com/k3s-io/k3s/pull/2897):  Fixed a typo in the README.md
* [#2141](https://github.com/k3s-io/k3s/pull/2141):  Updated Readme
* [#1935](https://github.com/k3s-io/k3s/pull/1935):  Add ability to disable kubeproxy

## Community PRs Opened
* [#3659](https://github.com/k3s-io/k3s/pull/3659):  Finding a best fit acronym explanation

**Comment by carloscarnero** [comment]
I'm finding these weekly statuses really nice! Kudos to the team!

**Reply by cwayne18** [reply]
Thank you! Let us know if there's something else you'd like to see added here, we're open to suggestions!

**Comment by cwayne18** [comment]
Updated, had accidentally left the default limit of 30 issues/pr's
